### ISSUE TITLE
bug(oxc-transform): Isolated Declarations Incorrectly Preserves Default Values in Rest Parameters

### ISSUE THREAD

**Issue by longlho** [oxc-project/oxc#17598]
## Description

When using `oxc-transform`'s isolated declarations feature (`typescript.declaration: true`), default values in rest parameters are incorrectly preserved in the generated `.d.ts` files. This causes TypeScript compiler errors when these declaration files are consumed, as parameter initializers are not allowed in declaration files.

## Environment

- **oxc-transform version**: Latest (as of 2026-01-02)
- **TypeScript version**: 5.x
- **tsconfig.json**: `isolatedDeclarations: true`

### Source Code (`dateTime.ts`)

```typescript
export function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...[value, options = {}]: Parameters<SomeFormatter['formatDate']>
): string {
  // implementation
}
```

### oxc-transform Configuration

```javascript
import { transformSync } from 'oxc-transform';

const result = transformSync(filePath, code, {
  lang: 'ts',
  sourcemap: false,
  typescript: {
    onlyRemoveTypeImports: true,
    declaration: true,
  },
});

// result.declaration contains the generated .d.ts
```

### Generated Declaration File (Incorrect)

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...[value, options = {}]: Parameters<SomeFormatter["formatDate"]>
): string;
```

### Expected Declaration File (Correct)

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  ...args: Parameters<SomeFormatter["formatDate"]>
): string;
```

Or better yet, the rest parameter should be expanded:

```typescript
export declare function formatDate(
  config: {
    locale: string;
    timeZone?: string;
  },
  getDateTimeFormat: SomeType,
  value: Parameters<SomeFormatter["formatDate"]>[0],
  options?: Parameters<SomeFormatter["formatDate"]>[1]
): string;
```

## Error

When consuming the generated `.d.ts` file, TypeScript reports:

```
error TS2371: A parameter initializer is only allowed in a function or constructor implementation.
```

## Impact

This bug prevents using oxc-transform's isolated declarations feature as a drop-in replacement for `tsc --emitDeclarationOnly` when the source code contains rest parameters with default values.

## Workaround

The only workaround is to modify the source code to avoid rest parameter syntax with defaults:

```typescript
// Instead of:
...[value, options = {}]: Parameters<SomeFormatter['formatDate']>

// Use explicit parameters:
value: Parameters<SomeFormatter['formatDate']>[0],
options: Parameters<SomeFormatter['formatDate']>[1] = {}
```

## Context

This issue was discovered while migrating a large monorepo ([formatjs](https://github.com/formatjs/formatjs)) to use oxc-transform for both JavaScript transpilation and declaration generation. The rest parameter pattern with defaults is relatively common in TypeScript codebases.

## Related

- TypeScript's `isolatedDeclarations` mode should not preserve implementation details like default values in declaration files
- The rest parameter syntax is a convenient way to maintain type safety with variadic functions
- This affects any TypeScript project using isolated declarations with oxc-transform

## Expected Behavior

oxc-transform's isolated declarations should either:
1. Strip default values from rest parameters in generated `.d.ts` files, OR
2. Expand rest parameters into individual parameters without default values in the declaration

---

**Sources:**
- [oxc-transform npm package](https://www.npmjs.com/package/oxc-transform)
- [Oxc Transformer Alpha announcement](https://oxc.rs/blog/2024-09-29-transformer-alpha)
- [TypeScript Transformer docs](https://oxc.rs/docs/guide/usage/transformer/typescript)

**Comment by Dunqing** [comment]
Fixed in #17602
